### PR TITLE
Add experience bar to user profiles

### LINKED ISSUE #2372: New profiles don't have level progression bar
old website has this
image
new website only shows the current level, and not the progress within it
image

### PR DESCRIPTION
**Pull request by LiquidPL**
Fixes #2372.

### CODE REVIEW

File: resources/assets/coffee/react/profile-page/stats.coffee [hunk 1]
```diff
@@ -57,7 +58,14 @@ ProfilePage.Stats = ({stats}) ->
...
             osu.transChoice 'common.count.second_short_unit', playtime.seconds()
-      div className: 'profile-badge profile-badge--level',
+      div className: 'profile-stats__stat-box profile-stats__stat-box--experience-bar',
+        div className: 'bar bar--user-profile',
+          div
+            className: 'bar__fill'
+            style:
+              width: "#{stats.level.progress}%"
+        div className: 'profile-stats__value profile-stats__value--level-progress', "#{stats.level.progress}%"
+      div className: 'profile-badge profile-badge--level profile-stats__badge',
```
Comment (nanaya) [inline resources/assets/coffee/react/profile-page/stats.coffee]: no mixing block and element.

File: resources/assets/coffee/react/profile-page/stats.coffee [hunk 2]
```diff
@@ -42,7 +42,8 @@ ProfilePage.Stats = ({stats}) ->
 
   div className: 'profile-stats',
     div className: 'profile-stats__row profile-stats__row--compact profile-stats__row--playtime',
-      div className: 'profile-stats__playtime-box',
+      div className: 'profile-stats__hover-target'
```
Comment (nanaya) [inline resources/assets/coffee/react/profile-page/stats.coffee]: should be the badge itself with flex order 1.

File: resources/assets/less/bem/profile-stats.less [hunk 3]
```diff
@@ -95,5 +126,10 @@
       text-align: center;
       padding-top: 10px;
     }
+
+    &--level-progress {
+      letter-spacing: 1px;
```
Comment (nanaya) [inline resources/assets/less/bem/profile-stats.less]: ?

File: resources/assets/less/bem/profile-stats.less [hunk 4]
```diff
@@ -46,10 +60,27 @@
...
+
+    &--playtime {
+      .@{_top}__hover-target:hover ~ & {
+        display: none;
+      }
+    }
+
+    &--experience-bar {
+      padding-right: 15px;
+      width: 190px;
```
Comment (nanaya) [inline resources/assets/less/bem/profile-stats.less]: `flex: 1 0 auto`

File: resources/assets/less/bem/profile-stats.less (hunk 3 above)
Comment (LiquidPL) [reply resources/assets/less/bem/profile-stats.less]: it feels like it's more spaced out than the other things in the design

**Pull request merged**